spec/support/file_system.rb

```diff
@@ -43,6 +37,23 @@ def chdir_and_allow_nesting(path)
       Dir.chdir(original_path)
     end
   end
+
+  private
+
+  def switch_home_directory(&block)
```

Comment:
We need something more complete than stubbing for the integration specs.

Even if our Ruby code was hitting a fake file system, any Git commands that are called in the integration specs would still read the user's real `.gitconfig` file from their real home directory. Since the contents of the `.gitconfig` can make a big difference to the output of Git commands, it would be hard to write tests that make assertions about output if the output was different on different systems.